api/python/provisioner/commands/generate_roster.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import logging
+from typing import Type
+from pathlib import Path
+
+from ..utils import dump_yaml
+from .. import (
+    inputs,
+    config,
+    values
+)
+from ..vendor import attr
+
+from ..pillar import PillarKey, PillarResolver
+
+from . import (
+    CommandParserFillerMixin
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class RunArgsGenerateRosterAttrs:
+    roster_path: str = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "roster file path"
+            }
+        },
+        default=str(config.PRVSNR_FACTORY_PROFILE_DIR / 'srv/roster')
+    )
+    user: str = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "user for salt-ssh"
+            }
+        },
+        default='root'
+    )
+    port: int = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "port for salt-ssh"
+            }
+        },
+        converter=int,
+        default=22
+    )
+    priv_key: str = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "private key path for salt-ssh"
+            }
+        },
+        default=str(config.PRVSNR_FACTORY_PROFILE_DIR / '.ssh/setup.id_rsa')
+    )
+
+
+@attr.s(auto_attribs=True)
+class GenerateRoster(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = RunArgsGenerateRosterAttrs
+
+    @staticmethod
+    def _get_pillar_data(key):
```

Comment:
Should not be a static method.
If we need a static method, let's create a utility class for all such methods that could be reusable for other classes.

Reply:
Because of codacy issue I have kept that methods as static otherwise codacy will raise issue for this.

can we ignore such a kind of codacy issue ?

Reply:
Yes, raise a ticket for such codacy issues with RE.
We can ignore them if they are genuine features.

Reply:
comment is addressed